### PR TITLE
Update lib/history.js to ES2015

### PR DESCRIPTION
**Pull request by Nathan-Schwartz**
All tests pass and the file runs in Node without babel. The linter passes for this file, aside from rules would have me change existing variable names (no-underscore-dangle, no-shadow). Let me know if I should take care of those errors as well, I didn't want to jump the gun.

### CODE REVIEW

File: lib/history.js [hunk 1]
```diff
@@ -1,190 +1,191 @@
...
+   */
+
+  _init() {
+    if (!this._storageKey) {
+      return;
+    }
+
+    // Load history from local storage
+    const persistedHistory = JSON.parse(this._localStorage.getItem(this._storageKey));
+    if (_.isArray(persistedHistory)) {
```
Comment (milesj) [inline lib/history.js]: This block can be rewritten.

```
if (Array.isArray(persistedHistory)) {
  this._hist.push(...persistedHistory);
}
```

File: lib/history.js [hunk 2]
```diff
@@ -1,190 +1,191 @@
...
+
+  /**
+   * Called when entering a mode
+   */
+
+  enterMode() {
+    // Reassign the command history to a
+    // cache, replacing it with a blank
+    // history for the mode.
+    this._histCache = _.clone(this._hist);
```
Comment (milesj) [inline lib/history.js]: We should be able to spread or concat to clone.

`[...this._hist]`

**Comment by Nathan-Schwartz** [comment]
@milesj Good to know, will do. Updated 👍

File: lib/history.js [hunk 3]
```diff
@@ -1,190 +1,190 @@
...
+
+  _init() {
+    if (!this._storageKey) {
+      return;
+    }
+
+    // Load history from local storage
+    const persistedHistory = JSON.parse(this._localStorage.getItem(this._storageKey));
+    if (Array.isArray(persistedHistory)) {
+      Array.prototype.push.apply(this._hist, persistedHistory);
```
Comment (milesj) [inline lib/history.js]: Should spread the push here.

**Comment by Nathan-Schwartz** [comment]
@milesj Good catch, updated

**Comment by milesj** [comment]
`+167 −167`. Nice!

**Pull request merged**